searx/engines/google_images.py

```diff
@@ -61,30 +61,25 @@ def request(query, params):
 
 # get response from search-request
 def response(resp):
+    results = []
+
     dom = html.fromstring(resp.text)
 
-    results = []
-    for element in dom.xpath('//div[@id="search"] //td'):
-        link = element.xpath('./a')[0]
-
-        google_url = urlparse(link.xpath('.//@href')[0])
-        query = parse_qs(google_url.query)
-        source_url = next(iter(query.get('q', [])), None)
-
-        title_parts = element.xpath('./cite//following-sibling::*/text()')
-        title_parts.extend(element.xpath('./cite//following-sibling::text()')[:-1])
-
-        result = {
-            'title': ''.join(title_parts),
-            'content': '',
-            'template': 'images.html',
-            'url': source_url,
-            'img_src': source_url,
-            'thumbnail_src': next(iter(link.xpath('.//img //@src')), None)
-        }
-
-        if not source_url or not result['thumbnail_src']:
+    # parse results
+    for result in dom.xpath('//div[@data-ved]'):
+
+        try:
+            metadata = loads(''.join(result.xpath('./div[contains(@class, "rg_meta")]/text()')))
+        except:
             continue
 
-        results.append(result)
+        # append result
+        results.append({'url': metadata['ru'],
```

Comment:
it would be better to move this part just below ```metadata = ....``` : if there is a problem getting metadata, the engine doesn't crash.